pkg/crypto/ed25519.go

```diff
@@ -0,0 +1,101 @@
+package crypto
+
+import (
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"io/ioutil"
+	"os"
+
+	"golang.org/x/crypto/ed25519"
+)
+
+var (
+	// PubKeyIdentifier is the PEM public key identifier
+	PubKeyIdentifier = "PUBLIC KEY"
+	// PrivKeyIdentifier is the PEM private key identifier
+	PrivKeyIdentifier = "PRIVATE KEY"
+	// PEMCipher is the PEM encryption algorithm
+	PEMCipher = x509.PEMCipherAES256
+	// PubKeyFilePermissions are the public key file perms
+	PubKeyFilePermissions os.FileMode = 0644
+	// PrivKeyFilePermissions are the private key file perms
+	PrivKeyFilePermissions os.FileMode = 0600
+)
+
+// LoadPublicKeyFromFile loads PEM formatted ED25519 public key from file.
+func LoadPublicKeyFromFile(publicKeyPath string) ([]byte, error) {
+	x509PEM, err := ioutil.ReadFile(publicKeyPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Parse x509 PEM file
+	block, _ := pem.Decode(x509PEM)
```

Comment:
shouldn't you loop through the various blocks and try to decode  all of them until you find a PubKeyIdentifier?

Reply:
In theory yes but normally only one key should be inside a PEM encoded keyfile :)

Reply:
So should we enforce that there is only one key in the PEM?